rt/test/TESTCMD3.MLC

```diff
@@ -0,0 +1,305 @@
+         STM   R14,R1,12(R13)      Save altered registers
+         MVC   W1Msg,0(R1)         Copy message to WTO
+         UNPK  DW(9),16(5,R13)     Convert RC to printable hex
+         TR    DW,H2P              Finish conversion
+         MVC   W1RC,DW             Copy to WTO
+         LA    R1,WTO1             R1 --> WTO
+         WTO   MF=(E,(1))          Show message and return code
+         LM    R14,R1,12(R13)      Restore altered registers
+         BR    R14                 Return to caller
+*
+***********************************************************************
+*        Set command to run -- Windows or Linux
+*
+*        ENVFlag has been set to Windows or Linux
+*
+*        R13 -> usable save area
+*        R14 =  return address
+*        R15 =  return code; always 0
+*
+***********************************************************************
+*
+SetCmd   DS    0H
+         STM   R14,R12,12(R13)     Save caller's registers
+*
+*        WARNING: no save area
+*
+         CLI   ENVFlag,ENVWin      Windows environment?
+         BE    SCWin               Yes; set Windows command
+*                                  No; must be Linux/Mac
+         LA    R0,TestCmdL         Linux/Mac test command
+         ST    R0,TestCmd@         Save the address
+         B     SCExit              All done
+SCWin    DS    0H
+         LA    R0,TestCmdW         Windows test command
+         ST    R0,TestCmd@         Save the address
+*NSI     B     SCExit              All done
+SCExit   DS    0H
+         SR    R15,R15             Set return code; always zero
+*
+         L     R14,12(,R13)        Restore caller's registers
+         LM    R0,R12,20(R13)      ... except for R15
+         BR    R14                 Return to caller
+*
+***********************************************************************
+*        Get runtime environment (set ENVFlag)
+*
+*        R13 -> usable save area
+*        R14 =  return address
+*        R15 =  return code; 0 = env found; 4 = unable to find env
+*
+***********************************************************************
+*
+GetEnv   DS    0H
+         STM   R14,R12,12(R13)     Save caller's registers
+*
+*        WARNING: no save area
+*
+         MVI   ENVFlag,ENVUnkn     Unknown environment
+*
+         GETENV COMSPEC            Check for Windows
+         LTR   R15,R15             Found COMSPEC?
+         BNZ   GE1000              No; check for Linux/Mac
+         MVI   ENVFlag,ENVWin      Yes; Indicate Windows
+*     LA    R3,63(,R2)
+*     SNAP STORAGE=((2),(3)),PDATA=,TEXT='Windows COMSPEC'
+         B     GE7000              All done; free GETENV storage
+GE1000   DS    0H
+***********************************************************************
+*        SHELL was not found in Github build environment (Linux)
+*        so the build failed. For now, assume environment is
+*        Linux/Mac if not Windows.
+***********************************************************************
```

Comment:
My view (and happy to discuss) is that we should only have "working" code in our source base. 
My limited understanding of assembler tells me that the linux check logic is still here and is just bypassed.
Personally I think it should be removed if it is not used. Just adds more complexity to the program that serves no purpose.
I am also interested for my own understand in the macro items in the code - Line 213 AGO .skplin and .skplin ANOP. This module is not a macro so does this remove these lines from the actual assembled program? What macro statements can you use outside of macros? Will this assemble using HLASM?

Reply:
Comment lines 204-208 were added to remind me and anyone modifying the code in the future why the Linux check was removed. At some future date someone might want to add a better check rather than just assume Linux if not Windows. A new, different, operating system that runs Java might turn up one day and a check for it might be needed.

The "AGO   .skplin" and ".skpllin ANOP" lines are allowed outside macros. It is a common technique to skip over code. Yes, the code will not not be assembled, and yes, the code would assemble using HLASM. (Well, not quite. I don't think the z390-specific macro GETENV is in any  IBM macro library, so a "macro not found" would result under HLASM.)

Try assembling the following:

```
FRED     CSECT
         SR    15,15
         AGO   .skp
         LA    15,4
         You won't see this line
.skp     ANOP
         BR    14
         END
```

In the resulting .PRN file you will see the AGO and ANOP but not the lines between them.

Removing the comment and the lines between the TESTCMD3 AGO and ANOP is not a problem. However, they do explain to a future maintainer why they are there and why they were skipped. Maybe that maintainer will think of a better way to add code back.

If you still want the lines removed, I will do so, in both TESTCMD3 and TESTCMD4. I will also remove lines 261-264 in TESTCMD3:

```
SHELL    DC    C'SHELL',X'00'      Linux/MacOS environment variable
*
*        Sample: SHELL=/bin/bash
*
```
and the corresponding lines in TESTCMD4.

Reply:
I'm ok to leave it. Does not hurt anything it's just clutter and I personally don't like that but understand how it provides clues to future maintainers.